### PR TITLE
Add complaint maintenance banner and upgrade Django-Flags

### PR DESCRIPTION
**Pull request by willbarton**
This PR adds a feature flag, a template, and a template tag for a complaint intake maintenance banner to sit on `/complaint` URLs. This is similar to #4745 and the service degradation banner, except for planned maintenance windows. 

The maintenance banner and its corresponding feature flag, `COMPLAINT_INTAKE_MAINTENANCE`, are intended to be used with Django-Flags "after date" and "before date" conditions to define the time window in which maintenance will take place. The banner uses the "before date" condition to inform users when the maintenance window will close and the complaint intake forms will be available again.

In doing this I've also simplified the implementation of the `COMPLAINT_INTAKE_TECHNICAL_ISSUES` flag to use required conditions in Django-Flags (https://github.com/cfpb/django-flags/pull/24) to require a particular matching path instead of requiring a separate setting of paths to match.

## Testing

1. Install the Django-Flags required-conditions branch in the Docker python container:
  
    ```shell
    ./shell.sh
    pip install git+https://github.com/cfpb/django-flags.git@required-conditions
   ```

2. Visit the flags admin at http://localhost:8000/admin/flags/

3. Add the following conditions:

   A required condition that will enable the flag after 2019/02/01 at midnight:

   - Flag: `COMPLAINT_INTAKE_MAINTENANCE`
   - Condition name: after date
   - Expected value: 2019-02-01T05:00Z
   - Required: Check the box

   A required condition that will disable the flag after 2019/03/01 at midnight:

   - Flag: `COMPLAINT_INTAKE_MAINTENANCE`
   - Condition name: before date
   - Expected value: 2019-03-01T05:00Z
   - Required: Check the box

   And finally a non-required condition that enable the flag if the required conditions are met:

   - Flag: `COMPLAINT_INTAKE_MAINTENANCE`
   - Condition name: boolean
   - Expected value: True
   - Required: DO NOT Check the box

4. Visit http://localhost:8000/complaint/ and see the banner:

image

## Notes:

This PR includes a non-required boolean condition set to `False` for both the complaint banners. Because the paths are now required feature flag conditions if it doesn't include a condition that disables it *with* the path, the banners will always appear. Clearly this is not desirable. Because the boolean condition is not required, when we add conditions that *do* enable to flag, they will be met and the flag will be enabled. The boolean condition merely achieves a default "off" state.

## Checklist

- [x] PR has an informative and human-readable title
- [x] Changes are limited to a single goal (no scope creep)
- [x] Code can be automatically merged (no conflicts)
- [ ] Code follows the standards laid out in the [CFPB development guidelines](https://github.com/cfpb/development)
- [x] Passes all existing automated tests
- [x] Any _change_ in functionality is tested
- [x] New functions are documented (with a description, list of inputs, and expected output)
- [x] Placeholder code is flagged / future todos are captured in comments
- [ ] Visually tested in supported browsers and devices (see checklist below :point_down:)
- [ ] Project documentation has been updated
- [x] Reviewers requested with the [Reviewers tool](https://help.github.com/articles/requesting-a-pull-request-review/) :arrow_right:

### CODE REVIEW

**Comment by willbarton** [comment]
PR is on hold for release of the Django-Flags changes as Django-Flags 4.1.0.

**Comment by willbarton** [comment]
Django-Flags 4.1.0 is released and this PR's tests can run and succeeded! It's ready for review.

File: cfgov/cfgov/settings/base.py [hunk 1]
```diff
@@ -599,84 +599,98 @@
...
-    },
+    'ASK_CFPB_H1': [
+        ('in split testing cluster', 'ASK_CFPB_H1')
+    ],
 
     # Test financial well-being hub pages on Beta
-    'FINANCIAL_WELLBEING_HUB': {'environment is': 'beta'},
+    'FINANCIAL_WELLBEING_HUB': [('environment is', 'beta')],
+
+    'MY_FLAG': [],
```
Comment (Scotchester) [inline cfgov/cfgov/settings/base.py]: Accidental addition?

File: cfgov/jinja2/v1/_includes/organisms/complaint-maintenance-banner.html [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+<div class="m-global-banner">
+  <div class="wrapper
+              wrapper__match-content">
+        {{ notification.render(
+            'warning',
+            true,
+            'We are currently undergoing planned maintenance.',
+            'Please come back '
+            + date_str + 
+            ' to submit or check the status of an existing complaint.'
```
Comment (Scotchester) [inline cfgov/jinja2/v1/_includes/organisms/complaint-maintenance-banner.html]:
```suggestion
            ' to submit a complaint or check the status of an existing complaint.'
```

Comment (willbarton) [reply cfgov/jinja2/v1/_includes/organisms/complaint-maintenance-banner.html]: Can we add this to the language that was already cleared?

File: cfgov/cfgov/settings/base.py (hunk 1 above)
Comment (willbarton) [reply cfgov/cfgov/settings/base.py]: Indeed it was!

File: docs/feature-flags.md [hunk 3]
```diff
@@ -184,9 +184,9 @@ FLAGS = {
     # Beta banner, seen on beta.consumerfinance.gov
     # When enabled, a banner appears across the top of the site proclaiming
     # "This beta site is a work in progress."
-    'BETA_NOTICE': {
-        'boolean': True,
-    },
+    'BETA_NOTICE': [
+        ('boolean', True),
```
[the remark this answers is not in the document]
Comment (willbarton) [reply docs/feature-flags.md]: In this document, probably so.

File: cfgov/jinja2/v1/_includes/organisms/complaint-maintenance-banner.html (hunk 2 above)
Comment (Scotchester) [reply cfgov/jinja2/v1/_includes/organisms/complaint-maintenance-banner.html]: IMO, yes, this is a non-substantive grammatical correction.

File: docs/feature-flags.md [hunk 4]
```diff
@@ -1,6 +1,6 @@
 # Feature flags
 
-Feature flags are implemented using our [Wagtail-Flags](https://github.com/cfpb/wagtail-flags) app. The [README](https://github.com/cfpb/wagtail-flags/blob/master/README.md) contains an overview and examples of how to use feature flags in Wagtail.
+Feature flags are implemented using our [Django-Flags](https://github.com/cfpb/django-flags) app. The [README](https://github.com/cfpb/wagtail-flags/blob/master/README.md) contains an overview and examples of how to use feature flags in Wagtail.
```
[the remark this answers is not in the document]
Comment (willbarton) [reply docs/feature-flags.md]: Better wording in 1a45e94.

File: docs/feature-flags.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (willbarton) [reply docs/feature-flags.md]: Switched in 1a45e94.

File: cfgov/v1/templatetags/complaint_banners.py [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+                before_date = dateparse.parse_datetime(before_dates[0])
+            except TypeError:
+                before_date = before_dates[0]
+
+            before_date = timezone.localtime(before_date)
+
+            if before_date > now:
+                date_str = 'after {time} on {date}'.format(
+                    time=format(before_date, 'g:i a e'),
+                    date=format(before_date, 'l, F j, Y'),
```
[the remark this answers is not in the document]
Comment (willbarton) [reply cfgov/v1/templatetags/complaint_banners.py]: Included in d0b31fa.

File: cfgov/cfgov/settings/base.py (hunk 1 above)
Comment (willbarton) [reply cfgov/cfgov/settings/base.py]: Removed in b0a8732.

File: cfgov/v1/templatetags/complaint_banners.py [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+
+@register.simple_tag
+def complaint_issue_banner(request):
+    template = 'organisms/complaint-issue-banner.html'
+    return mark_safe(render_to_string(template))
+
+
+@register.simple_tag
+def complaint_maintenance_banner(request):
+    """ Add a complaint maintenance banner. This banner is intended to be used
```
[the remark this answers is not in the document]
Comment (willbarton) [reply cfgov/v1/templatetags/complaint_banners.py]: Reformatted in [d0b31fa](https://github.com/cfpb/cfgov-refresh/commit/d0b31fad4a4020aaab7a1a9baa2ae4d1195675dc).

**Pull request merged**